app/controllers/shipping_api_controller.rb

```diff
@@ -0,0 +1,64 @@
+class ShippingApiController < ApplicationController
+  skip_before_filter :verify_authenticity_token, only: [:rates, :create_log]
+  before_action :set_shipping_rate, only: :rates
+
+  def rates
+    if @shipping_rate.valid?
+      estimates = @shipping_rate.ups_rates + @shipping_rate.usps_rates
+      render json: estimates, status: :ok # TODO: add status code tests
+    else
+      render json: @shipping_rate.errors, status: :bad_request # TODO: add status code tests
+
+    end
+  end
+
+  def create_log
```

Comment:
I do. I think it's better to separate concerns — even if the concerns are very small — because then you (a) know where to find things and (b) have a path for growth.